pkg/api/v1/conversion.go

```diff
@@ -510,17 +512,13 @@ func Convert_api_Pod_To_v1_Pod(in *api.Pod, out *Pod, s conversion.Scope) error
 			return err
 		}
 		out.Annotations[PodInitContainersAnnotationKey] = string(value)
-	} else {
-		delete(out.Annotations, PodInitContainersAnnotationKey)
```

Comment:
hrm, without this else block if someone sets the annotation for these objects but not the field, then the out object might still have init containers.  I think we can consider that coder error though - no one internally should be setting those annotations.